plugins/ModuleLevelTrigger.cpp

```diff
@@ -342,444 +195,55 @@ ModuleLevelTrigger::do_resume(const nlohmann::json& /*resumeobj*/)
-    m_bitword_check = check_trigger_bitwords();
-  }
-
-  if ((!m_use_bitwords) || (m_bitword_check)) {
-
-    dfmessages::TriggerDecision decision = create_decision(pending_td);
-
+    TLOG() << "Received decision with timestamp "
+             << decision.trigger_timestamp ;
+    
     if ((!m_paused.load() && !m_dfo_is_busy.load())) {
-
-      TLOG_DEBUG(3) << "Sending a decision with triggernumber " << decision.trigger_number << " timestamp "
+      TLOG_DEBUG(1) << "Sending a decision with triggernumber " << decision.trigger_number << " timestamp "
              << decision.trigger_timestamp << " start " << decision.components.front().window_begin << " end " << decision.components.front().window_end
- 	     << " number of links " << decision.components.size()
-             << " based on TC of type "
-             << static_cast<std::underlying_type_t<decltype(pending_td.contributing_tcs[m_earliest_tc_index].type)>>(
-                  pending_td.contributing_tcs[m_earliest_tc_index].type);
+ 	     << " number of links " << decision.components.size();
 
-      using namespace std::chrono;
+      //using namespace std::chrono;
       // uint64_t end_lat_prescale = duration_cast<nanoseconds>(system_clock::now().time_since_epoch()).count();
       try {
-        auto td_sender = get_iom_sender<dfmessages::TriggerDecision>(m_td_output_connection);
-        td_sender->send(std::move(decision), std::chrono::milliseconds(1));
+        m_decision_output->send(std::move(decision), std::chrono::milliseconds(1));
         m_td_sent_count++;
         m_new_td_sent_count++;
-        m_td_sent_tc_count += pending_td.contributing_tcs.size();
+//        m_td_sent_tc_count += pending_td.contributing_tcs.size();
         m_last_trigger_number++;
-        add_td(pending_td);
+//        add_td(pending_td);
       } catch (const ers::Issue& e) {
         ers::error(e);
-        TLOG_DEBUG(1) << "The network is misbehaving: it accepted TD but the send failed for "
-                      << pending_td.contributing_tcs[m_earliest_tc_index].time_candidate;
+        TLOG_DEBUG(1) << "The network is misbehaving: TD send failed for "
+                      << m_last_trigger_number;
         m_td_queue_timeout_expired_err_count++;
-        m_td_queue_timeout_expired_err_tc_count += pending_td.contributing_tcs.size();
+        //m_td_queue_timeout_expired_err_tc_count += pending_td.contributing_tcs.size();
       }
 
     } else if (m_paused.load()) {
       ++m_td_paused_count;
-      m_td_paused_tc_count += pending_td.contributing_tcs.size();
-      TLOG_DEBUG(1) << "Triggers are paused. Not sending a TriggerDecision for pending TD with start/end times "
-                    << pending_td.readout_start << "/" << pending_td.readout_end;
+      //m_td_paused_tc_count += pending_td.contributing_tcs.size();
+      TLOG_DEBUG(1) << "Triggers are paused. Not sending a TriggerDecision for TD with timestamp and type "
+                    << ts << "/" << tt;
```

Comment:
The type is a bitword (there can be multiple TCs of different types in TD), and I'm not sure if `auto` appreciates it, or just converts e.g. `000100010` (say, prescale & Timing types in this TD) to e.g. 34 (meaningless to reader).

Doesn't need to be resolved in this PR though, though maybe a TODO comment could be useful.